dusty/commands/script.py

```diff
@@ -24,18 +26,15 @@ def execute_script(app_name, script_name, script_arguments=[]):
     app_specs = get_specs()['apps'].get(app_name)
     if not app_specs:
         raise KeyError('No app found named {} in specs'.format(app_name))
-    script_spec = None
+    found_spec = False
     for script_dict in app_specs['scripts']:
         if script_dict['name'] == script_name:
-            script_spec = script_dict
-    if script_spec is None:
+            found_spec = True
+            break
+    if not found_spec:
         raise KeyError('No script found named {} in specs for app {}'.format(script_name, app_name))
 
-    base_script_command = '; '.join(script_spec['command'])
-    if script_arguments == []:
-        script_string = base_script_command
-    else:
-        script_string = '{} {}'.format(base_script_command, ' '.join(script_arguments))
+    command_file = '{}/{}'.format(container_code_path(app_specs), dusty_command_file_name(app_specs.name, script_name=script_name))
 
     container_name = get_dusty_container_name(app_name)
-    utils.exec_docker('exec', '-ti', container_name, 'sh', '-c', script_string)
+    utils.exec_docker('exec', '-ti', container_name, 'sh', command_file, *script_arguments)
```

Comment:
-c is for reading from a string, this is just specifying to run a file with script arguments